pkg/cli/scan.go

```diff
@@ -351,43 +424,75 @@ func resolveInputFileFromArg(inputFilePath string) (*os.File, error) {
-	cmd.Flags().StringVar(&p.localDBFilePath, "local-file-grype-db", "", "import a local grype db file")
-	cmd.Flags().StringVarP(&p.outputFormat, "output", "o", "", fmt.Sprintf("output format (%s), defaults to %s", strings.Join(validOutputFormats, "|"), outputFormatOutline))
-	cmd.Flags().BoolVarP(&p.sbomInput, "sbom", "s", false, "treat input(s) as SBOM(s) of APK(s) instead of as actual APK(s)")
-	cmd.Flags().BoolVar(&p.packageBuildLogInput, "build-log", false, "treat input as a package build log file (or a directory that contains a packages.log file)")
-	cmd.Flags().StringVar(&p.distro, "distro", "wolfi", "distro to use during vulnerability matching")
-	cmd.Flags().StringVarP(&p.advisoryFilterSet, "advisory-filter", "f", "", fmt.Sprintf("exclude vulnerability matches that are referenced from the specified set of advisories (%s)", strings.Join(scan.ValidAdvisoriesSets, "|")))
-	cmd.Flags().StringSliceVarP(&p.advisoriesRepoDirs, "advisories-repo-dir", "a", nil, "local directory for advisory data")
-	cmd.Flags().BoolVar(&p.disableSBOMCache, "disable-sbom-cache", false, "don't use the SBOM cache")
-	cmd.Flags().BoolVar(&p.triageWithGoVulnCheck, "govulncheck", false, "EXPERIMENTAL: triage vulnerabilities in Go binaries using govulncheck")
-	_ = cmd.Flags().MarkHidden("govulncheck") //nolint:errcheck
-}
+		nameMatches := lo.Filter(apkindex.Packages, func(pkg *repository.Package, _ int) bool {
+			return pkg != nil && pkg.Name == input
+		})
 
-const (
-	outputFormatOutline = "outline"
-	outputFormatJSON    = "json"
-)
+		if len(nameMatches) == 0 {
+			return nil, fmt.Errorf("no Wolfi package found with name %q in arch %q", input, arch)
+		}
 
-var validOutputFormats = []string{outputFormatOutline, outputFormatJSON}
+		vers := lo.Map(nameMatches, func(pkg *repository.Package, _ int) string {
+			return pkg.Version
+		})
 
-type inputScanResult struct {
-	InputFile string
-	Result    scan.Result
+		sort.Sort(versions.ByLatestStrings(vers))
+		latestVersion := vers[0]
+
+		var latestPkg *repository.Package
+		for _, pkg := range nameMatches {
+			if pkg.Version == latestVersion {
+				latestPkg = pkg
+				break
+			}
+		}
+		downloadURL := fmt.Sprintf("%s/%s/%s", apkRepositoryURL, arch, latestPkg.Filename())
+
+		tmpFile, err := os.CreateTemp("", fmt.Sprintf("%s-%s-%s-*.apk", arch, input, latestVersion))
+		if err != nil {
+			return nil, fmt.Errorf("creating temp dir: %w", err)
+		}
+		req, err := http.NewRequestWithContext(ctx, http.MethodGet, downloadURL, nil)
+		if err != nil {
+			return nil, fmt.Errorf("creating request for %q: %w", downloadURL, err)
+		}
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			return nil, fmt.Errorf("downloading %q: %w", downloadURL, err)
+		}
+
+		if resp.StatusCode != http.StatusOK {
+			return nil, fmt.Errorf("downloading %q (status: %d): %w", downloadURL, resp.StatusCode, err)
+		}
+		_, err = io.Copy(tmpFile, resp.Body)
```

Comment:
This is great! Curious if it would make sense to just pull out the SBOM here from the stream and just save the SBOM, and then not scan it above? At least I think it has the option for taking in an SBOM and using it.

Reply:
Also, if you do take the SBOM, from the looks of it, you could then use the reader and not have temp files, is that right?

Reply:
(following up from offline) We can't use the embedded SBOMs for scanning yet, but if/when we can, that's going to be really great for this flow. Also the new [govulncheck functionality](https://github.com/wolfi-dev/wolfictl/pull/450) needs access to raw Go binary bytes today, until we work out a way to get that scanner to take SBOM data directly.

Reply:
Just to close this out. These are different SBOMs :)